### PR TITLE
kubeadm: Replace *clientset.Clientset with clientset.Interface

### PR DESCRIPTION
**Pull request by luxas**
**What this PR does / why we need it**:

Needed for https://github.com/kubernetes/kubernetes/pull/48899
We should always use `clientset.Interface` instead of `*clientset.Clientset`, for better testability and all the other benefits of using an interface.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
Should be straightforward to merge

**Release note**:

```release-note
NONE
```
@timothysc @dmmcquay @pipejakob

### CODE REVIEW

File: cmd/kubeadm/app/cmd/token.go [hunk 1]
```diff
@@ -193,7 +193,7 @@ func NewCmdTokenGenerate(out io.Writer) *cobra.Command {
 }
 
 // RunCreateToken generates a new bootstrap token and stores it as a secret on the server.
-func RunCreateToken(out io.Writer, client *clientset.Clientset, token string, tokenDuration time.Duration, usages []string, description string) error {
+func RunCreateToken(out io.Writer, client clientset.Interface, token string, tokenDuration time.Duration, usages []string, description string) error {
```
Comment (timothysc) [inline cmd/kubeadm/app/cmd/token.go]: fwiw I explicitly call it kubernetes.Interface vs. using the internal vernacular b/c it's easier for new folks to understand.

**Review by timothysc — COMMENTED** [review]
It's not a blocker but I like the kubernetes.Interface naming.

**Review by MHBauer — APPROVED** [review]
yay generic interfaces!

File: cmd/kubeadm/app/cmd/token.go (hunk 1 above)
Comment (luxas) [inline cmd/kubeadm/app/cmd/token.go]: SGTM, we should probably rename a couple of other imports as well, like kubeadmconstants to just constants etc. However, not a priority for me rn

**Pull request merged**